### ISSUE TITLE
Vite v4 config types incompatible

### ISSUE THREAD

**Issue by JoostKersjes** [vitest-dev/vitest#2474]
### Describe the bug

When upgrading to Vite `4.0.0` running a TypeScript check will now think that the `test` config for vitest is invalid. This is the TS error in question:

```
error TS2345: Argument of type '{ plugins: Plugin_2[]; resolve: { alias: { "@": string; }; }; server: { port: number; }; test: { globals: boolean; environment: string; include: string[]; }; build: { target: string; }; }' is not assignable to parameter of type 'UserConfigExport'.
  Object literal may only specify known properties, and 'test' does not exist in type 'UserConfigExport'.

 24   test: {
      ~~~~~~~
 25     globals: true,
    ~~~~~~~~~~~~~~~~~~
... 
 27     include: ["**/__tests__/*.test.ts"],
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 28   },
```

### Reproduction

https://stackblitz.com/edit/vitest-dev-vitest-1gphuh?file=vite.config.ts

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-11370H @ 3.30GHz
    Memory: 8.08 GB / 15.31 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 16.18.0 - ~/.nvm/versions/node/v16.18.0/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v16.18.0/bin/npm
  Browsers:
    Chromium: 108.0.5359.94
    Firefox: 107.0.1
  npmPackages:
    @vitejs/plugin-vue: 3.2.0 => 3.2.0 
    vite: 4.0.0 => 4.0.0 
    vitest: 0.25.6 => 0.25.6
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by christopherschroer** [comment]
The PR #2470 appears to address this

**Comment by sheremet-va** [comment]
Fixed by #2470
Released as part of 0.25.7
